libraries/Microsoft.Bot.Builder.Core.Extensions/TimeoutWarningMiddleware.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    public class TimeoutWarningMiddleware : IMiddleware
+    {
+        /// <summary>
+        /// Period of time to wait before logging a warning to indicate a potential channel timeout before the bot responds. Defaults to 8000ms.
+        /// </summary>
+        private readonly int _timeoutPeriod;
+
+        public TimeoutWarningMiddleware(int timeoutPeriod = 8000)
```

Comment:
/// Comments, including params. Suggest "timeoutPeriodMs" as the name so folks can tell the unit.